winpr/libwinpr/crypto/cipher.c

```diff
@@ -500,44 +512,57 @@ mbedtls_cipher_type_t winpr_mbedtls_get_cipher_type(int cipher)
 }
 #endif
 
-BOOL winpr_Cipher_Init(WINPR_CIPHER_CTX* ctx, int cipher, int op, const BYTE* key, const BYTE* iv)
+WINPR_CIPHER_CTX* winpr_Cipher_New(int cipher, int op, const BYTE* key, const BYTE* iv)
 {
+	WINPR_CIPHER_CTX* ctx;
 #if defined(WITH_OPENSSL)
 	int operation;
 	const EVP_CIPHER* evp;
+	EVP_CIPHER_CTX* octx;
+#elif defined(WITH_MBEDTLS)
+	int key_bitlen;
+	mbedtls_operation_t operation;
+	mbedtls_cipher_type_t cipher_type;
+	const mbedtls_cipher_info_t* cipher_info;
+#endif
+
+	ctx = calloc(1, sizeof(WINPR_CIPHER_CTX));
```

Comment:
I think we leak this calloc in case an error occurs later on in winpr_Cipher_New